api/v1beta1/index/cluster.go

```diff
@@ -28,8 +28,10 @@ import (
 )
 
 const (
```

Comment:
I would suggest to keep the existing ByClusterClassName / ClusterByClusterClassClassName funcs untouched because if someone outside of core CAPI is using them, this change will break them and it's very hard to notice.

Also we couldn't really backport this change to the index.

Let's instead add a new index with the following names:
* `ByClusterClassRef`
* `ClusterByClusterClassClassRef`
* `spec.topology.classRef`
* `ClusterClassKey` => `ClusterClassRef`